### DISCUSSION TITLE
Home moves caret to first non-whitespace character with Ideavim

### DISCUSSION THREAD

**Discussion by henry034** [JetBrains/ideavim#455] [Q&A]
I found that when the vim emulator is on, the first smart keys configuration does not work. (Home moves caret to first non-whitespace character). But other smart keys seem to be good (e.g. Insert paired brackets)
If this condition is common, how do I set the function myself?

**Comment by AlexPl292** [comment]
Hi, IdeaVim plugin overrides the behaviour of the home key. Seems like at the moment it's not possible to turn this off.  I've created an issue for that https://youtrack.jetbrains.com/issue/VIM-2560

**Comment by citizenmatt** [comment]
Would it help to add `nmap <Home> ^` to your `~/.ideavimrc`? This will map the Home key to `^`, which will move the caret to the first non-whitespace character in the line? (This doesn't help with pressing Home when you're already at the first non-whitespace character. In IntelliJ this will move to the start of the line. `0` in IdeaVim will do the same)
